spec/integration/inventory2_spec.rb

```diff
@@ -483,17 +488,119 @@ def fact_plan(name = 'facts_test')
+          name: conn[:host],
+          config: {
+            transport: conn[:protocol],
+            conn[:protocol] => {
+              host: conn[:host],
+              user: conn[:user],
+              port: conn[:port],
+              'host-key-check' => false
+            }
+          }
+        }]
       }
     end
 
-    let(:shell_cmd) { 'whoami' }
-
     it 'runs a plan' do
       result = run_cli_json(run_plan)
       expect(result).not_to include('kind')
       expect(result.length).to eq(1)
       expect(result[0]).to include('status' => 'success', 'target' => conn[:host])
     end
+
+    context 'applies config to dynamic inventory' do
+      context 'with name and uri set' do
+        let(:target) { 'myhostname' }
+        let(:facts) do
+          { conn[:host] => {
+            'fact1' => true,
+            'identity' => { 'user' => conn[:second_user] },
+            'uri_fact' => conn[:host],
+            'name_fact' => target
+          } }
+        end
+
+        let(:addtl_inv) do
+          { name: 'facts.name_fact',
+            uri: 'facts.uri_fact' }
+        end
+
+        let(:ssh_config) do
+          { user: 'facts.identity.user',
+            password: 'facts.identity.user' }
+        end
+
+        it 'runs a plan' do
+          result = run_cli_json(run_plan)
+          expect(result).not_to include('kind')
+          expect(result.length).to eq(1)
+          expect(result[0]).to include('status' => 'success', 'target' => 'myhostname')
+        end
+
+        it 'handles structured facts' do
+          result = run_cli_json(run_command)
+          expect(result).not_to include('kind')
+          expect(result['items'][0]['result']['stdout']).to eq("#{conn[:second_user]}\n")
+        end
+      end
+
+      context 'on another node' do
+        let(:target) { 'bullseye' }
+        let(:facts) do
+          { conn[:host] => {
+            'uri_fact' => conn[:host],
+            'name_fact' => target,
+            'fact1' => true
+          } }
+        end
+
+        let(:addtl_inv) do
+          { name: 'facts.name_fact',
+            uri: 'facts.uri_fact' }
+        end
+
+        it 'uses fact-based name' do
+          result = run_cli_json(run_command)
+          expect(result).not_to include('kind')
+          expect(result['items'][0]['target']).to eq(target)
+        end
+      end
+
+      context 'when a fact is not set' do
+        let(:facts) do
+          { conn[:host] => {
+            'fact1' => true
+          } }
+        end
+
+        let(:ssh_config) { { user: 'facts.identity.user' } }
+
+        it 'falls back to default' do
+          result = run_cli_json(run_command)
+          expect(result).not_to include('kind')
+          expect(result['items'][0]['result']['stdout']).to eq("#{conn[:user]}\n")
+        end
+      end
+
+      context 'using certname fact' do
+        # TODO
+      end
+
+      context 'on a non-queried node' do
+        # Are curly braces : rspec :: parens : lisps?
```

Comment:
I think you need the "double" parens here, the outer set is the block, the inner set is a hash.

Reply:
Wait is this something I actually need to change? The comment was just meant to be cheeky.

Reply:
I couldn't tell if you were actually asking about the curly braces. It looks right if you want that memoized helper to return a hash.